www/class/CentreonLDAPAdmin.class.php

```diff
@@ -167,30 +194,37 @@ public function setGeneralOptions(array $options, $arId = 0)
                     "LDAP PARAM - Warning the reference date wasn\'t set for LDAP : " . $options['ar_name']
                 );
             }
-            $this->db->query(
-                "INSERT INTO auth_ressource (ar_name, ar_description, ar_type, ar_enable, ar_sync_base_date) 
-                VALUES ('" . $this->db->escape($options['ar_name']) . "',
-                    '" . $this->db->escape($options['ar_description']) . "',
-                    'ldap',
-                    '" . $options['ldap_auth_enable']['ldap_auth_enable'] . "',
-                    '" . $options['ar_sync_base_date'] . "')"
+            $statement = $this->db->prepare(
+                "INSERT INTO auth_ressource (ar_name, ar_description, ar_type, ar_enable, ar_sync_base_date)
+                VALUES (:ar_name, :ar_description,'ldap', :ar_enable, :ar_sync_base_date)"
             );
-            $maxArIdSql = "SELECT MAX(ar_id) as last_id
-                          FROM auth_ressource
-                          WHERE ar_name = '" . $this->db->escape($options['ar_name']) . "'";
-            $res = $this->db->query($maxArIdSql);
+            $statement->bindValue(':ar_name', $options['ar_name'], \PDO::PARAM_STR);
+            $statement->bindValue(':ar_description', $options['ar_description'], \PDO::PARAM_STR);
+            $statement->bindValue(':ar_enable', $options['ldap_auth_enable']['ldap_auth_enable']);
+            $statement->bindValue(':ar_sync_base_date', $options['ar_sync_base_date'], \PDO::PARAM_INT);
+            $statement->execute();
+            $maxArIdSql = "SELECT MAX(ar_id) as last_id FROM auth_ressource WHERE ar_name = :ar_name";
+            $res = $this->db->prepare($maxArIdSql);
+            $res->bindValue(':ar_name', $options['ar_name'], \PDO::PARAM_STR);
+            $res->execute();
             $row = $res->fetch();
             $arId = $row['last_id'];
             unset($res);
         } else {
-            $this->db->query(
+            $statement = $this->db->prepare(
                 "UPDATE auth_ressource
-                SET ar_name = '" . $this->db->escape($options['ar_name']) . "',
-                ar_description = '" . $this->db->escape($options['ar_description']) . "',
-                ar_enable = '" . $options['ldap_auth_enable']['ldap_auth_enable'] . "',
-                ar_sync_base_date = '" . $options['ar_sync_base_date'] . "'
-                WHERE ar_id = " . $this->db->escape($arId)
+                SET ar_name = :ar_name,
+                ar_description = :ar_description,
+                ar_enable = :ar_enable,
+                ar_sync_base_date = :ar_sync_base_date
+                WHERE ar_id = :ar_id"
             );
+            $statement->bindValue(':ar_name', $options['ar_name'], \PDO::PARAM_STR);
```

Comment:
same

Reply:
it's already sanitized on form.php line 339 ($value used on setGeneralOption called in line 395 on form.php)